source/contentRecog/recogUi.py

```diff
@@ -171,6 +177,7 @@ def _recogOnResult(result):
 		queueHandler.queueFunction(queueHandler.eventQueue,
 			ui.message, _("Recognition failed"))
 		return
-	resObj = RecogResultNVDAObject(result=result)
-	# This method queues an event to the main thread.
-	resObj.setFocus()
+	# Ensure a copy of the recognizer exists and call it's getResultHandler method to present the result
+	# the returned handler returned can be used for any further processing.
+	if recognizer:
+		recognizer.getResultHandler(result)

```

Comment:
'getResultHandler' creates an instance of the result handler, but none of the classes take ownership of it? 

Maybe `getResultHandler` should be `showResult` and recognizer creates, and keeps the instance internally?

Reply:
While working on this you mentioned that returning an object of the `ResultHandlerClass` would potentially be helpful for future NVDA developers as it would give them access to the result. This would allow them to add features like result caching etc. For now, this isn't required and so I omitted storing the resultHandler object in a variable that is never used.